src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
+    let i = 0;
+
+    if (arguments.length < 2) {
+      i = 1;
+      prev = this[0];
+    }
+
+    for (i; i < this.length; i++) {
+      prev = callback(prev, this[i], i, this);
+    }
+
+    return prev;
```

Comment:
Let's improve your solution a bit
```suggestion
    let prev = initialValue;
    let startIndex = 0;

    if (arguments.length < 2) {
      startIndex = 1;
      prev = this[0];
    }

    for (let i = startIndex; i < this.length; i++) {
      prev = callback(prev, this[i], i, this);
    }

    return prev;
```